command/agentproxyshared/cache/lease_cache.go

```diff
@@ -235,7 +234,9 @@ func (c *LeaseCache) checkCacheForRequest(id string, req *SendRequest) (*SendRes
 	if token != "" {
 		// This is a static secret check. We need to ensure that this token
 		// has previously demonstrated access to this static secret.
-		if !slices.Contains(index.Tokens, token) {
+		// We could check the capabilities cache here, but since these
+		// indexes should be in sync, this saves us an extra cache get.
+		if _, ok := index.Tokens[token]; !ok {
```

Comment:
It doesn't look like this map access is protected by IndexLock, aren't we at risk of a concurrent write to it?

Reply:
Hmm, you're right, yeah. We should hold a read lock here. Let me fix that

Reply:
Added! Good call out :)